spyder/plugins/editor/widgets/tests/test_hints_and_calltips.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+"""Tests for editor calltips and hover hints tooltips."""
+
+# Standard library imports
+import sys
+
+# Third party imports
+from qtpy.QtCore import Qt, QPoint
+from qtpy.QtGui import QTextCursor
+import pytest
+
+# Constants
+PY2 = sys.version[0] == '2'
+TEST_SIG = 'some_function(hello=None)'
+TEST_DOCSTRING = "This is the test docstring."
+TEST_TEXT = """'''Testing something'''
+def {SIG}:
+    '''{DOC}'''
+
+
+some_function""".format(SIG=TEST_SIG, DOC=TEST_DOCSTRING)
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize('params', [
+            # Parameter, Expected Output
+            ('dict', 'dict'),
+            ('type', 'type'),
+            (TEST_TEXT, TEST_SIG)
+        ]
+    )
+def test_get_calltips(qtbot, lsp_codeeditor, params):
+    """Test that the editor is returning hints."""
+    code_editor, _ = lsp_codeeditor
+
+    param, expected_output_text = params
+
+    # Set text in editor
+    code_editor.set_text(param)
+
+    code_editor.moveCursor(QTextCursor.End)
+
+    with qtbot.waitSignal(code_editor.sig_signature_invoked,
+                          timeout=30000) as blocker:
+        code_editor.document_did_change('')
+        qtbot.keyPress(code_editor, Qt.Key_ParenLeft, delay=1000)
+        qtbot.wait(3000)
+
+        # Not working on some CI
+        # assert code_editor.calltip_widget.isVisible()
+
+        signatures = blocker.args[0]['params']['signatures']
+        assert expected_output_text in signatures['label']
+
+
+# @pytest.mark.slow
+@pytest.mark.parametrize('params', [
+            # Parameter, Expected Output
+            ('dict', '' if PY2 else 'dict'),
+            ('type', 'type'),
+            ('import math', 'module'),
+            (TEST_TEXT, TEST_DOCSTRING)
+        ]
+    )
+def test_get_hints(qtbot, lsp_codeeditor, params):
+    """Test that the editor is returning hover hints."""
+    code_editor, _ = lsp_codeeditor
+
+    param, expected_output_text = params
+
+    # Set text in editor
+    code_editor.set_text(param)
+
+    code_editor.moveCursor(QTextCursor.End)
+    x, y = code_editor.get_coordinates('cursor')
+
+    # The `- 5` is to put the mouse on top of the word
+    point = code_editor.calculate_real_position(QPoint(x - 5, y))
+
+    with qtbot.waitSignal(code_editor.sig_display_object_info,
+                          timeout=30000) as blocker:
+        code_editor.hide_tooltip()
```

Comment:
This should be put before the `with`, right?

Reply:
This might not be needed anymore now that I am checking the output of the signal and not the visibility of the hover hints